### PR TITLE
feat: Cloudflare captcha integrated

### LINKED ISSUE #9044: [CAL-1739] ability to add captcha to booking page to prevent spam
a user is reporting being booked by bots

we should consider adding some sort of captcha / rate limiting to prevent spam bookings

<sub>[CAL-1739](https://linear.app/calcom/issue/CAL-1739/ability-to-add-captcha-to-booking-page-to-prevent-spam)</sub>

### PR DESCRIPTION
**Pull request by Ninad3721**
## What does this PR do?
 I have integrated cloudflare turnstile in  the booking page, which will block bots from booking someone.

Fixes https://github.com/calcom/cal.com/issues/9044

 It disables confirm button until the cloudflare turnstile test is passed
image

Then after getting test getting solved the confirm button is enabled 
image

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

1.Click on three dots beside profile
2.Click on view public profile
3.Book a slot

## Checklist

**special note** : After running yarn test-e2e some test regarding booking are failed , which I think because of adding captcha as these test are conducted by bots (I think so).

### CODE REVIEW

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 1]
```diff
@@ -659,11 +661,19 @@ const BookingPage = ({
                       ? "-mt-4"
                       : ""
                   )}>
+                <Turnstile
+                    className="fixed bottom-4 right-4"
+                    onError={() => setStatus("error")}
+                    onExpire={() => setStatus("expired")}
+                    onSuccess={() => setStatus("solved")}
+                    siteKey="0x4AAAAAAAFM2dMLqlOd35hp"
```
Comment (baileypumfleet) [inline apps/web/components/booking/pages/BookingPage.tsx:669]: You've exposed the site key here, this should be in a .env file

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 2]
```diff
@@ -659,11 +661,19 @@ const BookingPage = ({
                       ? "-mt-4"
                       : ""
                   )}>
+                <Turnstile
+                    className="fixed bottom-4 right-4"
+                    onError={() => setStatus("error")}
```
Comment (baileypumfleet) [inline apps/web/components/booking/pages/BookingPage.tsx:666]: It looks like we're not using the error and expire success states

**Review by pumfleet — CHANGES_REQUESTED** [review]
Left a couple of comments, but the ESLint issues need to be fixed too

Thank you for your contribution!

File: apps/web/components/booking/pages/BookingPage.tsx (hunk 2 above)
Comment (Ninad3721) [reply apps/web/components/booking/pages/BookingPage.tsx:666]: Ok, @baileypumfleet  I will change these and I have to make a new PR right ? Also I used lint:fix and that fixed all linting errors, I think this the right way to do it. Sorry for the silly questions actually I am new to open source contributions.

**Review by PeerRich — CHANGES_REQUESTED** [review]
hey, great work, however we need to make this optional. not everyone wants this. can you instead make a standalone "Cloudflare Captcha" App and launch it in the app store? here is how: https://cal.com/docs/how-to-guides/how-to-build-an-app#steps-in-creating-an-app

File: apps/web/components/booking/pages/BookingPage.tsx (hunk 1 above)
Comment (PeerRich) [reply apps/web/components/booking/pages/BookingPage.tsx:669]: ideally this should actually be added in the App Config (see for example how the GTM app works or the Fathom App). You would make an event-type App and then add your key in the input box

Comment (PeerRich) [reply apps/web/components/booking/pages/BookingPage.tsx:669]: @hariombalhara we should maybe write an abstraction/adapter to enable all kinds of captchas  the same way we offer all kinds of analytics apps

**Comment by Ninad3721** [comment]
> hey, great work, however we need to make this optional. not everyone wants this. can you instead make a standalone "Cloudflare Captcha" App and launch it in the app store? here is how: https://cal.com/docs/how-to-guides/how-to-build-an-app#steps-in-creating-an-app

So Should I start working on it ??

**Comment by Ninad3721** [comment]
> > hey, great work, however we need to make this optional. not everyone wants this. can you instead make a standalone "Cloudflare Captcha" App and launch it in the app store? here is how: https://cal.com/docs/how-to-guides/how-to-build-an-app#steps-in-creating-an-app
> 
> So Should I start working on it by creating an issue  ??

**Comment by hariombalhara** [comment]
@Ninad3721 yeah please do

**Pull request closed without merging**